src/overlays/actors/ovl_En_Am/z_en_am.c

```diff
@@ -89,81 +124,453 @@ static DamageTable D_808B10C8 = {
+    func_808B0208(this, globalCtx);
+    if (Actor_XZDistanceToPoint(&this->actor, &this->actor.home.pos) < 8.0f) {
+        func_808B04E4(this);
+    }
+}
+
+void func_808B0640(EnAm* this) {
+    this->unk_236 = 0x28;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->unk_240 = 0.0f;
+    this->actionFunc = func_808B066C;
+}
+
+void func_808B066C(EnAm* this, GlobalContext* globalCtx) {
+    if (this->unk_236 != 0) {
+        this->unk_236--;
+    } else {
+        this->unk_23C = this->actor.yawTowardsPlayer;
+        func_808B0208(this, globalCtx);
+        if (this->unk_23C == this->actor.shape.rot.y) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_AMOS_VOICE);
+            func_808B0358(this);
+        }
+    }
+}
+
+void func_808B06D0(EnAm* this, GlobalContext* globalCtx) {
+    Animation_Change(&this->skelAnime, &D_06005B3C, 1.0f, 4.0f, (Animation_GetLastFrame(&D_06005B3C) - 6), 2, 0.0f);
+    func_800BE504(&this->actor, &this->enemyCollider);
+    this->actor.speedXZ = 6.0f;
+    func_800BCB70(&this->actor, 0x4000, 0xFF, 0, (Animation_GetLastFrame(&D_06005B3C) - 0xA));
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_EYEGOLE_DAMAGE);
+    this->enemyCollider.base.acFlags &= ~1;
+    this->unk_234 = 0xFF;
+    this->actionFunc = func_808B07A8;
+}
+
+void func_808B07A8(EnAm* this, GlobalContext* globalCtx) {
+    Math_StepToF(&this->actor.speedXZ, 0.0f, 0.5f);
+    if (SkelAnime_Update(&this->skelAnime) != 0) {
+        if (this->actor.colChkInfo.health == 0) {
+            func_808B0820(this);
+        } else {
+            this->enemyCollider.base.acFlags |= AC_ON;
+            func_808B0358(this);
+        }
+    }
+}
+
+void func_808B0820(EnAm* this) {
+    Animation_PlayLoopSetSpeed(&this->skelAnime, &D_06000238, 4.0f);
+    this->unk_236 = 64;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actor.flags |= 0x10;
+    this->actor.speedXZ = 0.0f;
+    this->unk_240 = 6.0f;
+    this->actionFunc = func_808B0894;
+}
+
+void func_808B0894(EnAm* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f dustPos;
+    s32 pad;
+
+    this->unk_236 -= 1;
+    this->unk_23C = this->actor.yawTowardsPlayer;
+    func_808B0208(this, globalCtx);
+    if (this->unk_236 == 1) {
+        EnBom* bomb = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.world.pos.x,
+                                          this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 2, 0);
+        if (bomb != NULL) {
+            bomb->timer = 0;
+        }
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_EYEGOLE_DEAD);
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0xB0);
+
+        for (i = 0; i < 8; i++) {
+            dustPos.x = (Math_SinS(0) * 7.0f) + this->actor.world.pos.x;
+            dustPos.y = (randPlusMinusPoint5Scaled(10.0f) * 6.0f) + (this->actor.world.pos.y + 40.0f);
+            dustPos.z = (Math_CosS(0) * 7.0f) + this->actor.world.pos.z;
```

Comment:
Isn't `Math_SinS(0)` just 0 and isn't `Math_CosS(0)` just 1? Seems odd but okay

Reply:
Yeah it was an odd match, I had that initially, but it wanted the sin and cos for matching 🤷